example_workspace/src/pluginlib_sample_app/jni/src/main/cpp/polygon_loader.cpp

```diff
@@ -89,5 +81,24 @@ int main()
     LOGD("The plugin failed to load for some reason. Error: %s", ex.what());
   }
 
-  return 0;
+  while(1) {
+    int events;
+    struct android_poll_source* source;
+
+    // Poll android events, without locking
+    while (ALooper_pollAll(0, NULL, &events, (void**)&source) >= 0) {
+        // Process this event
+        if (source != NULL) {
+            source->process(state, source);
+        }
+
+        // Check if we are exiting.
+        if (state->destroyRequested != 0) {
+            LOGD("APP DESTROYED BYE BYE");
+            return;
+        }
+    }
```

Comment:
No timeout and no `ros::Rate`. Busy-wait?

Reply:
Yes. I will change for a locking looper version (ALooper_pollAll(-1, ...)), as nothing more is done.